scripts/build/Jenkins/Jenkinsfile

```diff
@@ -384,6 +360,190 @@ def HandleDriveMount(String snapshot, String repositoryName, String projectName,
+            controlFile.ENABLE_GEMS.each { gemName ->
+                PlatformSh("scripts/o3de${ext} enable-gem --gem-path ${workspace}/${repoName}/${gemName} --project-name AutomatedTesting || ver>nul", "Enabling ${workspace}/${repoName}/${gemName} for AutomatedTesting")
+            }
+        }
+    }
+
+    //check out all repos
+    if(controlFile.REPOS) {
+        echo 'REPOS'
+        controlFile.REPOS.each { repo ->
+            //stop cyclical repos
+            if(PROCESSED_REPOS.contains(repo.NAME)) {
+                echo "Already processed ${repo.NAME}"
+                return
+            }
+
+            def repoControlFile = {}
+            def foundRepoControlFile = false
+            dir(repo.NAME) {
+                CheckoutRepo(repo.BRANCH, [GetRemoteConfig(repo.NAME, repo.URL)])
+
+                if(fileExists('.automatedtesting.json')) {
+                    echo "Loading ${repo.URL} .automatedtesting.json"
+                    repoControlFile = readJSON file: '.automatedtesting.json'
+                    foundRepoControlFile = true
+                }
+            }
+
+            if(foundRepoControlFile) {
+                //There is a control file, execute it
+                ExecuteControlFile(repoControlFile, repo.NAME)
+            } else {
+                //There is no control file so we must determine if this is a composite repo or a singular repo
+                //A standard composite repo has the form
+                //root
+                //  Engines
+                //  Projects
+                //  Gems
+                //  Templates
+                //  Restricted
+                def enginesExist = false
+                def projectsExist = false
+                def gemsExist = false
+                def templatesExist = false
+                def restrictedExist = false
+                def engineRepo = false
+                def projectRepo = false
+                def gemRepo = false
+                def templateRepo = false
+                def restrictedRepo = false
+
+                dir(repo.NAME) {
+                    if(fileExists("Engines")) {
+                        enginesExist = true
+                    }
+                    if(fileExists("Projects")) {
+                        projectsExist = true
+                    }
+                    if(fileExists("Gems")) {
+                        gemsExist = true
+                    }
+                    if(fileExists("Templates")) {
+                        templatesExist = true
+                    }
+                    if(fileExists("Restricted")) {
+                        restrictedExist = true
+                    }
+
+                    //If this is not a standard composite repo, then it wont have any composite dirs, then we know
+                    //its a singular object repo, see what kind it is.
+                    if(!enginesExist && !projectsExist && !gemsExist && !templatesExist && !restrictedExist) {
```

Comment:
No, the repo astom-sampleviewer is conformant because it is a singular repo that has children, like the engine and is not a compound repo like extras. In singular repos only the top level object needs top be linked and all children are known by the parent. In a compund repo like extras there is no parent object, as the repo.json is not a parent.